src/io/leopard/toLeopard.ts

```diff
@@ -1051,32 +1049,29 @@ export default function toLeopard(
 
           ${target.volume !== 100 ? `this.audioEffects.volume = ${target.volume};` : ""}
 
-          ${[...target.variables, ...target.lists]
-            .map(variable => `this.vars.${variable.name} = ${toOptimalJavascriptRepresentation(variable.value)};`)
-            .join("\n")}
-
           ${[...target.variables, ...target.lists]
             .map(
               variable =>
-                [
-                  variable,
-                  Object.entries(variableNameMap.get(target)).find(([, newName]) => newName === variable.name)[0]
-                ] as [Variable | List, string]
+                `this.vars.${variableNameMap[variable.id]} = ${toOptimalJavascriptRepresentation(variable.value)};`
             )
-            .filter(([variable, oldName]) => variable.visible || shownWatchers.has(oldName))
-            .map(([variable, oldName]) => {
-              return `this.watchers.${variable.name} = new Watcher({
-              label: ${JSON.stringify((target.isStage ? "" : `${target.name}: `) + oldName)},
+            .join("\n")}
+
+          ${[...target.variables, ...target.lists]
+            .map(variable => [variable, variableNameMap[variable.id]] as [Variable | List, string])
```

Comment:
Why this `map` here? The variable's name isn't used in the subsequent `filter` operation, so it looks like fetching the variable's name from `variableNameMap` could be moved into the function after the `filter`.

Reply:
Thanks, good catch. Previous logic had the following `filter` make use of the name (previously the *old* i.e. Scratch name), for checking in the "shown watchers" list. Your suggestion makes the intermediate array unneeded, which is a nice optimization (both in performance and conceptually)!